cpmpy/solvers/cplex.py

```diff
@@ -0,0 +1,459 @@
+        elif not self.license_ok():
+            raise Exception("CPM_cplex: A problem occured during license check. Make sure your installed the CPLEX Optimization Studio")
+        import docplex.mp.model as dmm
+
+        self.cplex_model = dmm.Model()
+        super().__init__(name="cplex", cpm_model=cpm_model)
+
+    @property
+    def native_model(self):
+        """
+            Returns the solver's underlying native model (for direct solver access).
+        """
+        return self.cplex_model
+
+
+    def solve(self, time_limit=None, **kwargs):
+        """
+            Call the cplex solver
+
+            Arguments:
+            - time_limit:  maximum solve time in seconds (float, optional)
+            - kwargs:      any keyword argument, sets parameters of solver object
+
+            Examples of supported arguments include:
+                - context (optional) – An instance of context to be used in instead of the context this model was built with.
+                - cplex_parameters (optional) – A set of CPLEX parameters to use instead of the parameters defined as context.cplex_parameters. Accepts either a RootParameterGroup object (obtained by cloning the model’s parameters), or a dict of path-like names and values.
+                - checker (optional) – a string which controls which type of checking is performed. Possible values are: - ‘std’ (the default) performs type checks on arguments to methods; checks that numerical arguments are numbers, but will not check for NaN or infinity. - ‘numeric’ checks that numerical arguments are valid numbers, neither NaN nor math.infinity - ‘full’ performs all possible checks, the union of ‘std’ and ‘numeric’ checks. - ‘off’ performs no checking at all. Disabling all checks might improve performance, but only when it is safe to do so.
+                - log_output (optional) – if True, solver logs are output to stdout. If this is a stream, solver logs are output to that stream object. Overwrites the context.solver.log_output parameter.
+                - clean_before_solve (optional) – a boolean (default is False). Solve normally picks up where the previous solve left, but if this flag is set to True, a fresh solve is started, forgetting all about previous solves..
+
+            For a full list of parameters, please visit https://ibmdecisionoptimization.github.io/docplex-doc/mp/docplex.mp.model.html?#docplex.mp.model.Model.solve
+            and for cplex parameters: https://www.ibm.com/docs/en/icos/22.1.1?topic=cplex-topical-list-parameters
+        """
+        # ensure all vars are known to solver
+        self.solver_vars(list(self.user_vars))
+
+        if time_limit is not None and not np.isinf(time_limit):
+            self.cplex_model.set_time_limit(time_limit)
+
+        cplex_objective = self.cplex_model.get_objective_expr()
+        self.cplex_model.solve(**kwargs)
+        # all available solve details:
```

Comment:
You mean a user can extract these statistics with `self.cplex_model.solve_details`? If so, might be good to make that explicit

Reply:
moved it to the docs of .solve, i guess that's the place where it makes most sense?